llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -2056,6 +2056,14 @@ void DwarfDebug::beginInstruction(const MachineInstr *MI) {
     }
   }
 
+  auto RecordSourceLine = [&](auto &DL, auto Flags) {
```

Comment:
I'd use `DebugLoc` instead of auto here. `DL` in LLVM also commonly used for data layout, so making the time explicit would be helpful here.